src/Components/OATGraphViewer/OATGraphViewer.tsx

```diff
@@ -952,33 +1049,27 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
     const onElementClick = (evt: Event, node: IOATNodeElement) => {
         if (!state.modified) {
             // Checks if a node is selected to display it in the property editor
-            if (node.data.type === OATInterfaceType && translatedOutput) {
-                currentNodeIdRef.current = node.id;
-
-                const currentModel = translatedOutput.find(
-                    (model) => model['@id'] === node.id
+            if (
+                node.data.type === OATInterfaceType &&
+                translatedOutput &&
+                node.id !== currentNodeIdRef.current // Prevent re-execute the same node
+            ) {
+                execute(
+                    () => {
+                        currentNodeIdRef.current = node.id;
+                        dispatch({
+                            type: SET_OAT_SELECTED_MODEL_ID,
+                            payload: node.id
+                        });
+                    },
+                    () => {
+                        const selectedModelIdCopy = selectedModelId;
```

Comment:
We probably don't need to duplicate this variable right? It'll be the same reference anyway?